kong/plugins/cors/handler.lua

```diff
@@ -56,8 +58,24 @@ end
 
 local function configure_credentials(ngx, conf)
   if ngx.ctx.cors_allow_all then
-    ngx.header["Access-Control-Allow-Credentials"] = "false"
+    local allow_credentials = false
+
+    -- If it's catch all, then match the Origin requested by the client
+    -- but only if the Origin URL is valid, and only if credentials=true
+    if conf.credentials then
+      local origin = req_get_headers()["Origin"]
+      if origin then
+        local parsed_url = url.parse(origin)
+        if parsed_url.scheme and parsed_url.host then
+          ngx.header["Access-Control-Allow-Origin"] = fmt("%s://%s", 
```

Comment:
In general, don't use uppercase. Openresty returns them all as lowercase. If a header is not found, some metatable magic kicks in converts to lowercase and searches again.

performance wise: always use lowercase